services/workflows-service/prisma/migrations/20250129142137_documents_init/migration.sql

```diff
@@ -0,0 +1,81 @@
+/*
+  Warnings:
+
+  - You are about to drop the column `documents` on the `Business` table. All the data in the column will be lost.
+
+*/
+-- CreateEnum
+CREATE TYPE "DocumentStatus" AS ENUM ('provided', 'unprovided', 'requested');
+
+-- CreateEnum
+CREATE TYPE "DocumentDecision" AS ENUM ('approved', 'rejected', 'revisions');
+
+-- CreateEnum
+CREATE TYPE "DocumentFileType" AS ENUM ('selfie', 'document', 'other');
+
+-- CreateEnum
+CREATE TYPE "DocumentFileVariant" AS ENUM ('front', 'back', 'other');
+
+-- AlterTable
+ALTER TABLE "Business" DROP COLUMN "documents";
+
+-- CreateTable
+CREATE TABLE "Document" (
+    "id" TEXT NOT NULL,
+    "category" TEXT NOT NULL,
+    "type" TEXT NOT NULL,
+    "issuingVersion" TEXT NOT NULL,
+    "version" INTEGER NOT NULL,
+    "status" "DocumentStatus" NOT NULL,
+    "decision" "DocumentDecision",
+    "properties" JSONB NOT NULL,
+    "businessId" TEXT,
```

Comment:
I don't think we raised this option in the planning discussion, but having a relationship table for each entity is probably the complete solution here, as it will preserve database constraints. 

on the other hand, doing the migration if the future to support it should be relatively straightforward

Reply:
For properties? How would we support the various schemas?

Reply:
no, for entities (business, enduser..)